server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -0,0 +1,92 @@
+package com.flipkart.varadhi.auth;
+
+import com.flipkart.varadhi.entities.UserContext;
+import com.flipkart.varadhi.exceptions.InvalidConfigException;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.flipkart.varadhi.auth.ResourceAction.ORG_CREATE;
+
+public class DefaultAuthorizationProvider implements AuthorizationProvider {
+    private DefaultAuthorizationConfiguration configuration;
+
+    @Override
+    public Future<Boolean> init(JsonObject configuration) {
+        this.configuration = configuration.mapTo(DefaultAuthorizationConfiguration.class);
+        return Future.succeededFuture(true);
+    }
+
+    /**
+     * Checks if the user {@code UserContext} is authorized to perform action {@code ResourceAction} on resource path.
+     *
+     * @param userContext information on the user, contains the user identifier
+     * @param action      the action being performed by the user which is to be authorized
+     * @param resource    the path of the resource on which the action is to be authorized
+     *
+     * @return {@code Future<Boolean>} a future result expressing True/False decision
+     */
+    @Override
+    public Future<Boolean> isAuthorized(UserContext userContext, ResourceAction action, String resource) {
+        // parse the resource path based on the action and return the final resourceIDs for each resourceType
+        Map<ResourceType, String> resourceTypeToResourceId = getResourceIds(action, resource);
```

Comment:
would it be more beneficial to evaluate it from root to leaf or opposite, thoughts ?

Reply:
Currently we iterate over map entries, so the evaluation order here is undefined.

One problem them I'm facing is that in case of project, how do I figure out the parent entity ID?

Some approaches I have in mind:

1. Pass only the project_id as resource (as being currently done in RequiredAuthorization) and the Authorization Provider has the required mapping to the parent entity. So, when we fetch the role_bindings for a given resource_id, we get the role_bindings at that resource_id level along with the role_bindings at the parent level till the root node.
2. Or, since this is a Default barebones impl, we just do away with the requirement of inheriting permissions and check only at the resource_ids being passed.

Now coming to the original question, what would be better? Root to Leaf or opposite?
I'd say, considering the points above, opposite evaluation will be much better as we are mostly being passed the last set of identifiers through which we will mostly move up the hierarchy. Evaluating Root to Leaf will involve the extra work of figuring out the entire hierarchy at once and then evaluating.

What are your thoughts?

Reply:
Updated the code, we are evaluating in lazy manner from leaf to root node